js/scenes/Item.js

```diff
@@ -1,66 +1,116 @@
 import React, { Component, PropTypes } from 'react';
+import { fetchItemIfNeeded } from '../store/actions';
+import { defaultTextStyle } from '../../config';
+import Header from '../components/Header';
+import { connect } from 'react-redux';
+
 import {
   View,
-  Text,
-  TouchableOpacity,
+  Image,
   StyleSheet,
+  ActivityIndicator,
 } from 'react-native';
-import { connect } from 'react-redux';
 
-import Header from '../components/Header';
-import { defaultTextStyle } from '../../config';
+import {
+  FETCHING,
+} from '../store/constants';
 
 export class Item extends Component {
   constructor(props) {
     super(props);
-
     this.navigator = this.props.navigator;
-    this.onMapPress = this.onMapPress.bind(this);
-    this.onBackPress = this.onBackPress.bind(this);
   }
 
-  onMapPress() {
-    this.navigator.push({ id: 'map' });
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  componentWillReceiveProps({ itemId }) {
+    if (itemId !== this.props.itemId) this.fetchData();
   }
 
-  onBackPress() {
-    this.navigator.pop();
+  fetchData() {
+    const { dispatch, itemId } = this.props;
+    dispatch(fetchItemIfNeeded(itemId));
   }
 
   render() {
     return (
-      <View>
+      <View style={styles.root}>
         <Header
           title="Item"
-          navigator={this.navigator}/>
-        <TouchableOpacity
-          onPress={this.onMapPress}>
-          <Text style={styles.text}>Go to Map scene</Text>
-        </TouchableOpacity>
+          navigator={this.navigator}
+          style={styles.header}/>
+        {this.renderContent()}
       </View>
     );
   }
+
+  renderContent() {
+    const { item } = this.props;
+    if (!item || item.status === FETCHING) return this.renderContentLoading();
+    const { value: { image }} = item;
+
+    return (
+      <Image
+        style={styles.image}
+        source={{ uri: image }}
+        resizeMode={Image.resizeMode.cover}
```

Comment:
According to the doc, the `resizeMode` accepts a string, so it should read `resizeMode="cover"`.
Maybe it has changed in the last few versions.